Add Notices Documentation to Storybook and upgrade to Storybook 7 (#11524)
woocommerce/woocommerce-gutenberg-products-block: assets/js/base/components/notice-banner/docs/docs.mdx

import { Canvas, Meta, ArgTypes, Primary, Source } from '@storybook/blocks';

import * as NoticeBannerStories from '../stories/index.stories.tsx';

<Meta name="Docs" of={ NoticeBannerStories } />

# NoticeBanner

An informational UI displayed near the top of the store pages.

<Primary />

## Design Guidelines

`NoticeBanner` is an informational UI element displayed near the top of store pages used to indicate the result of an action, or to draw the user's attention to necessary information.

Notices are color-coded to indicate the type of message being communicated, and also show an icon to reinforce the meaning of the message. The color and icon used for a notice are determined by the `status` prop.

### Default Notices

By default, noices are grey and used for less important messaging.

<Canvas of={ NoticeBannerStories.Default } />

### Informational Notices

> Blue notices with an info icon are used for general information for buyers that are not blocking and do not require action.

Reviewer: ```suggestion
Blue notices with an info icon are used for general information for buyers, but do not require them to take an action.
```
